aether-entity-extraction-module/extractor/manager.py

```diff
@@ -247,7 +234,29 @@ def push_to_kernel(self):
                             self.redis
                         )
                 except Exception as e:
-                    logger.debug(str(e))
+                    logger.error(str(e))
+
+            if current_entity_size:
+                entities = [self.PROCESSED_ENTITIES.pop() for _ in range(current_entity_size)]
+                # group entities by realm
+                entities_by_realm = {}
+                for entity in entities:
+                    realm = submission_tenant[entity['submission']]
+                    if realm not in entities_by_realm:
+                        entities_by_realm[realm] = []
+                    entities_by_realm[realm].append(entity)
```

Comment:
I long time ago a read that the pythonic way of programming is to force the exceptions, not trying to avoid them

```python
try:
    entities_by_realm[realm].append(entity)
except KeyError:
    entities_by_realm[realm] = [entity]
```

In this way you increase the execution speed (it only fails the first time and you don't check the realm condition in each iteration). I think one of the things we want to achieve with this "extractor" is high performance otherwise I would have said nothing. :wink: